com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -188,19 +188,19 @@ public int compare(SPIRVReflector.Resource a, SPIRVReflector.Resource b) {
         }
     }
 
-    private ShaderDesc.Shader.Builder compileGLSLToSPIRV(ByteArrayInputStream is, ES2ToES3Converter.ShaderType shaderType, IResource resource, String resourceOutput, String targetProfile, boolean isDebug, boolean soft_fail)  throws IOException, CompileExceptionError {
-        InputStreamReader isr = new InputStreamReader(is);
-
-        // Convert to ES3 (or GL 140+)
-        CharBuffer source = CharBuffer.allocate(is.available());
-        isr.read(source);
-        source.flip();
-
-        // TODO: Add optional glslangValidator step here to ensure shader is ES2 or Desktop X compliant?
+    static public class SPIRVCompileResult
+    {
+        public byte[] source;
+        public ArrayList<String> compile_warnings = new ArrayList<String>();
+        public ArrayList<SPIRVReflector.Resource> attributes = new ArrayList<SPIRVReflector.Resource>();
+        public ArrayList<SPIRVReflector.Resource> resource_list = new ArrayList<SPIRVReflector.Resource>();
+    };
 
-        ES2ToES3Converter.Result es3Result = ES2ToES3Converter.transform(source.toString(), shaderType, targetProfile);
+    static public SPIRVCompileResult compileGLSLToSPIRV(String shaderSource, ES2ToES3Converter.ShaderType shaderType, String resourceOutput, String targetProfile, boolean isDebug, boolean soft_fail)  throws IOException, CompileExceptionError {
```

Comment:
I think we can remove the soft_fail flag now and output proper errors for the SPIRV compilation, since it's opt-in functionality due to the new project setting.

Reply:
I agree!